core/src/main/java/org/elasticsearch/index/query/GeoDistanceRangeQueryBuilder.java

```diff
@@ -19,41 +19,60 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.search.Query;
+import org.elasticsearch.Version;
 import org.elasticsearch.common.geo.GeoDistance;
+import org.elasticsearch.common.geo.GeoHashUtils;
+import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.unit.DistanceUnit;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.fielddata.IndexGeoPointFieldData;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.geo.GeoPointFieldMapper;
+import org.elasticsearch.index.search.geo.GeoDistanceRangeQuery;
 
 import java.io.IOException;
 import java.util.Locale;
+import java.util.Objects;
 
 public class GeoDistanceRangeQueryBuilder extends AbstractQueryBuilder<GeoDistanceRangeQueryBuilder> {
 
     public static final String NAME = "geo_distance_range";
 
-    private final String name;
+    private final String fieldName;
 
     private Object from;
     private Object to;
     private boolean includeLower = true;
     private boolean includeUpper = true;
 
-    private double lat;
+    private Double lat;
 
-    private double lon;
+    private Double lon;
```

Comment:
I think we can move these two to be a GeoPoint now, and keep track of the breaking aspect in migrate_query_refactoring.asciidoc